### PR TITLE
Create NPZ file for training

### PR DESCRIPTION
**Pull request by ngreenwald**
Basic functionality for build.py to create an NPZ with specified overlap and crop size.

### CODE REVIEW

File: caliban_toolbox/build.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    """Add padding to an image to enable additional edge crops to be generated
+
+    Args:
+        images: array of images to crop
+        crop_size: size of crops that will be produced
+
+    Returns:
+        np.array: padded image
+    """
+    row_offset, col_offset = int(crop_size[0] / 2), int(crop_size[1] / 2)
```
[the remark this answers is not in the document]
Comment (ngreenwald) [reply caliban_toolbox/build.py:72]: Yup, we'll only ever be specifying powers of 2 for our final image size

File: caliban_toolbox/build.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    return padded_images
+
+
+def combine_npz_files(npz_list, resize_ratios, extra_edge_padding=False,
+                      stride_ratio=1, final_size=(256, 256)):
+    """Take a series of NPZ files and combine together into single training NPZ
+
+    Args:
+        npz_list: list of NPZ files to combine
+        resize_ratios: ratio used to resize each NPZ if data is of different resolutions
```
Comment (MekWarrior) [inline caliban_toolbox/build.py:88]: Is this meant to be a list or an array? does 1 indicate no scaling required?

**Comment by MekWarrior** [comment]
@ngreenwald I think we need to take another look at this PR. It seems somewhat fragile as presently constructed and heavily tailored to static image set. There also appears to be several locations in the code that call for padding on top of existing padding, is that by design?

File: caliban_toolbox/build.py (hunk 2 above)
Comment (ngreenwald) [reply caliban_toolbox/build.py:88]: Yes, this is a list that matches the NPZ files being specified, where 1 indicates no resizing required

**Comment by ngreenwald** [comment]
For sure, I actually have no idea what the NPZ file for training a tracking network looks like, so I'm guessing as is this won't work for that. Happy to get advice on how to make it more generalizable, or if we should explicitly restrict to static?

Yes, the two paddings accomplish different goals. The first padding is to make the full image size an even multiple of the desired crop size for the network. This is because the deepcell_toolbox.tile_image function will index from the end of the image to make up for any non-even division, which isn't the desired behavior since it will override the user-specified stride ratio. 

The second padding is for data augmentation: it optionally adds blank padding equal to half of a crop size on either side of the image to squeeze an extra crop out of each row/col of the image.

**Comment by ngreenwald** [comment]
Hey @MekWarrior, what do we need to do to merge this in?

File: requirements.txt [hunk 3]
```diff
@@ -11,3 +11,4 @@ boto3>=1.9.0
 xarray==0.12.1
 netCDF4==1.5.3
 pathlib==1.0.1
+deepcell-toolbox >= 0.4.0
```
Comment (MekWarrior) [inline requirements.txt:14]: Bump version and eliminate spaces

Comment (ngreenwald) [inline requirements.txt:14]:
```suggestion
deepcell-toolbox>=0.6.1
```

File: caliban_toolbox/build.py (hunk 2 above)
Comment (MekWarrior) [reply caliban_toolbox/build.py:67]: There should be a threshold for scaling. (e.g. 1.01 does not need to re-scaled)

File: caliban_toolbox/build.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    """
+
+    combined_x = []
+    combined_y = []
+
+    for idx, npz in enumerate(npz_list):
+        current_x = npz['X']
+        current_y = npz['y']
+
+        # resize if needed
```
Comment (ngreenwald) [inline caliban_toolbox/build.py:83]:
```suggestion
        # resize if needed
        # TODO: Add tolerance to control when resizing happens
```

File: caliban_toolbox/build.py [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+        new_images = np.zeros((images.shape[0], new_row_len, new_col_len, images.shape[3]))
+        new_images[:, :row_len, :col_len, :] = images
+        return new_images
+
+
+def combine_npz_files(npz_list, resize_ratios, stride_ratio=1, final_size=(256, 256)):
+    """Take a series of NPZ files and combine together into single training NPZ
+
+    Args:
+        npz_list: list of NPZ files to combine
```
Comment (ngreenwald) [inline caliban_toolbox/build.py:66]:
```suggestion
        npz_list: list of NPZ files to combine. Currently only works on 2D static data
```

**Pull request merged**